exanauts/ColPack.jl
# Upgrade ColPack bindings, options as strings, systematic tests

This PR supersedes #19

**Source - breaking**

- Version bump to 0.5.0
- Removed the very inefficient `matrix2adjmatrix`: non-symmetric matrices must be handled directly with bipartite coloring methods.
- The coloring method and order must now be given as strings (and not trivial structs). The list of accepted strings can be found here https://github.com/CSCsw/ColPack/tree/master/Examples, and user inputs are checked against said list, with informative errors.

**Source - not breaking**

- New version 0.4.0 of the ColPack_jll dependency
- Addition of `ColPackPartialColoring` and `ColPackBiColoring` in addition to `ColPackColoring`
- Only `ColPackColoring` and `ColPackPartialColoring` are exported, I still don't understand what `ColPackBiColoring` returns
- Partial column coloring of a CSC matrix is done with partial row coloring of its transpose, which is in the proper CSR format. **Unpleasant consequence: partial coloring from a matrix and coloring from a file give different results in general.**

**Documentation**

- Removed outdated tutorial and example
- Improve docstrings with code examples and lists of supported options

**Tests**

- Removed SparseDiffTools comparison (it may rely on implementation details anyway)
- Add systematic tests that every method and order is supported
- Add exception-throwing tests
- Right now there are no correctness checks, it's hard to do those with so many different coloring methods. Should I add some tests for specific colorings?
